modules/lily-compiler/src/test/java/io/github/tomboyo/lily/compiler/feature/ComposedSchemaTest.java

```diff
@@ -0,0 +1,108 @@
+package io.github.tomboyo.lily.compiler.feature;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.github.tomboyo.lily.compiler.LilyExtension;
+import io.github.tomboyo.lily.compiler.LilyExtension.LilyTestSupport;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+public class ComposedSchemaTest {
+  @Nested
+  @ExtendWith(LilyExtension.class)
+  class OneOf {
+    @BeforeAll
+    static void beforeAll(LilyTestSupport support) {
+      support.compileOas(
+          """
+            openapi: 3.0.2
+            components:
+                schemas:
+                  MySchema:
+                    oneOf:
+                      - $ref: '#/components/schemas/Foo'
+                      - type: object
+                        properties:
+                          foo:
+                            type: integer
+                            format: int32
+                      - type: string
+                      - type: string
+                        format: email
+                      - type: integer
+                        format: int32
+                  #MySchemaStringEmailAlias:
+                  #  type: string
+                  #  format: email
```

Comment:
Create an issue for this:
A OAS document could define an object schema component MySchemaStringEmailAlias and a OneOf composed schema MySchema with an anonymous email string member, resulting in a name conflict because both of those things become names MySchemaStringEmailAlias in the base package.

Reply:
https://github.com/Tomboyo/lily/issues/89